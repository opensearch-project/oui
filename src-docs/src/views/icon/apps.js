/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import {
  OuiFlexGrid,
  OuiFlexItem,
  OuiIcon,
  OuiPanel,
  OuiCopy,
  OuiCodeBlock,
  OuiSpacer,
} from '../../../../src/components';

const iconTypes = [
  'advancedSettingsApp',
  'apmApp',
  'appSearchApp',
  'auditbeatApp',
  'canvasApp',
  'codeApp',
  'consoleApp',
  'crossClusterReplicationApp',
  'dashboardApp',
  'devToolsApp',
  'discoverApp',
  'emsApp',
  'filebeatApp',
  'gisApp',
  'graphApp',
  'grokApp',
  'heartbeatApp',
  'indexManagementApp',
  'indexPatternApp',
  'indexRollupApp',
  'metricsApp',
  'lensApp',
  'logsApp',
  'machineLearningApp',
  'managementApp',
  'metricbeatApp',
  'monitoringApp',
  'notebookApp',
  'packetbeatApp',
  'pipelineApp',
  'recentlyViewedApp',
  'reportingApp',
  'savedObjectsApp',
  'searchProfilerApp',
  'securityAnalyticsApp',
  'securityApp',
  'spacesApp',
  'sqlApp',
  'timelionApp',
  'upgradeAssistantApp',
  'uptimeApp',
  'usersRolesApp',
  'visualizeApp',
  'watchesApp',
  'workplaceSearchApp',
];

export default () => (
  <>
    <OuiCodeBlock language="html" isCopyable paddingSize="m">
      {'<OuiIcon type="advancedSettingsApp" size="xl" />'}
    </OuiCodeBlock>
    <OuiSpacer />
    <OuiFlexGrid direction="column" columns={3}>
      {iconTypes.map((iconType) => (
        <OuiFlexItem key={iconType}>
          <OuiCopy
            display="block"
            textToCopy={iconType}
            afterMessage={`${iconType} copied`}>
            {(copy) => (
              <OuiPanel
                hasShadow={false}
                hasBorder={false}
                onClick={copy}
                paddingSize="s">
                <OuiIcon
                  className="oui-alignMiddle"
                  type={iconType}
                  size="xl"
                />{' '}
                &emsp; <small>{iconType}</small>
              </OuiPanel>
            )}
          </OuiCopy>
        </OuiFlexItem>
      ))}
    </OuiFlexGrid>
  </>
);
