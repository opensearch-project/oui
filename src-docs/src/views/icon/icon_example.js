/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { GuideSectionTypes } from '../../components';

import {
  OuiCode,
  OuiIcon,
  OuiToken,
  OuiLink,
  OuiText,
  OuiCallOut,
  OuiSpacer,
} from '../../../../src/components';

import iconConfig from './playground';

import Icons from './icons';

import Tokens from './tokens';

import CustomTokens from './custom_tokens';

import Apps from './apps';

import Editor from './editor';

import Ml from './ml';

import Logos from './logos';

import IconSizes from './icon_sizes';

import IconColors from './icon_colors';
import AppIconColors from './icon_colors_apps';

import IconTypes from './icon_types';
const iconTypesSource = require('!!raw-loader!./icon_types');

export const IconExample = {
  title: 'Icons',
  sections: [
    {
      text: (
        <p>
          <strong>OuiIcon</strong> is a handy component for using our custom
          glyphs and logos. The <OuiCode>type</OuiCode> prop accepts either an
          enumerated name from one of the sets below, a location to a custom SVG
          asset, or a React Element.
        </p>
      ),
      demo: <OuiIcon type="grid" />,
      props: { OuiIcon },
      playground: iconConfig,
    },
    {
      text: (
        <OuiCallOut
          iconType="accessibility"
          title={
            <>
              For better accessibility it&apos;s always recommended to give a
              descriptive <OuiCode>title</OuiCode> based on the icon use.
            </>
          }
          color="warning">
          <p>
            If no title is provided the icon is going to be purely decorative
            and it will get by default an{' '}
            <OuiCode language="js">aria-hidden=true</OuiCode>.
          </p>
        </OuiCallOut>
      ),
    },
    {
      title: 'Glyphs',
      text: (
        <>
          <p>
            Glyphs are small, monochromatic icons that typically should always
            use the default size of{' '}
            <OuiCode language="js">size=&quot;m&quot;</OuiCode>.
          </p>
        </>
      ),
      demo: <Icons />,
    },
    {
      title: 'Editor controls',
      text: (
        <p>
          Editor icons relate to the visual styling of elements and are commonly
          used within <strong>OuiButtonGroup</strong> components.
        </p>
      ),
      demo: <Editor />,
    },
    {
      title: 'OpenSearch logos',
      text: (
        <p>
          Product logos follow similar rules as app logos. Note the use of{' '}
          <OuiCode>.ouiIcon__fillNegative</OuiCode> on portions of the SVGs to
          handle flipping colors for dark mode.
        </p>
      ),
      demo: <Logos />,
    },
    {
      title: 'Apps',
      text: (
        <p>
          App logos are usually displayed at <OuiCode>32x32</OuiCode> or above
          and can contain multiple colors.
        </p>
      ),
      demo: <Apps />,
    },
    {
      text: (
        <>
          <h3>Machine learning icons</h3>
          <p>
            Machine learning has some specific icons for job creation. Again,
            these are made for <OuiCode>32x32</OuiCode>.
          </p>
        </>
      ),
      demo: <Ml />,
    },
    {
      title: 'Tokens',
      text: (
        <>
          <p>
            Tokens are most commonly used to visually signify field or code
            types. An <strong>OuiToken</strong> accepts any valid{' '}
            <strong>OuiIcon</strong> as its <OuiCode>iconType</OuiCode>{' '}
            property. However, icons designed specifically for use in the{' '}
            <strong>OuiToken</strong> are prefixed with &quot;token&quot; in
            their name and have pre-defined styles.
          </p>
        </>
      ),
      props: { OuiToken },
      demo: <Tokens />,
    },
    {
      wrapText: false,
      text: (
        <>
          <OuiText>
            <h3>Custom tokens</h3>
            <p>
              By default, an <OuiCode>iconType</OuiCode> with the token prefix
              (i.e. those listed above) will have predefined styles. However,
              any valid <OuiCode>iconType</OuiCode> can be passed and, in either
              case, the <OuiCode>shape</OuiCode>, <OuiCode>size</OuiCode>,{' '}
              <OuiCode>color</OuiCode>, and <OuiCode>fill</OuiCode> can be
              customized.
            </p>
          </OuiText>
          <OuiSpacer />
          <CustomTokens />
        </>
      ),
    },
    {
      title: 'Sizes',
      text: (
        <p>
          Use the <OuiCode>size</OuiCode> prop to automatically size your icons.
          Medium is the default, and will output a <OuiCode>16x16</OuiCode>{' '}
          icon.
        </p>
      ),
      demo: <IconSizes />,
    },
    {
      title: 'Colors',
      text: (
        <p>
          The default behavior of icons is to inherit from the text color. You
          can use the <OuiCode>color</OuiCode> prop to assign a custom color
          which accepts a named color from our palette or a valid&nbsp;
          <OuiLink
            target="_blank"
            href="https://developer.mozilla.org/en-US/docs/Web/CSS/color_value">
            CSS color data type
          </OuiLink>
          &nbsp;which will be passed down through the inline-style{' '}
          <OuiCode>fill</OuiCode>&nbsp; property.{' '}
          <strong>We recommend relying on the OUI named color palette</strong>{' '}
          unless the custom color is initiated by the user (like as a graph
          color).
        </p>
      ),
      demo: <IconColors />,
    },
    {
      wrapText: false,
      text: (
        <>
          <OuiText>
            <p>
              Two-tone icons, like our app style icons, will behave similarly to
              normal glyphs when provided a specific color by applying the color
              to <strong>all</strong> the shapes within. You can force the icon
              to match the parent's text color by passing{' '}
              <OuiCode>color="inherit"</OuiCode> to the icon.
            </p>
          </OuiText>
        </>
      ),
      demo: <AppIconColors />,
    },
    {
      title: 'Custom SVGs',
      text: (
        <>
          <p>
            The <OuiCode>type</OuiCode> prop can accept a valid enum, string or
            React SVG Element. When using a custom SVG, please make sure it sits
            on a square canvas and preferably utilizes one of OUI&apos;s sizes (
            <OuiCode>16x16</OuiCode> or <OuiCode>32x32</OuiCode>).
          </p>
          <p>
            When using custom SVGs for simple glyphs,{' '}
            <strong>remove all fill attributes</strong> on the SVG and utilize
            the CSS helpers if you have complex logos that need to work with
            theming.
          </p>
        </>
      ),
      source: [
        {
          type: GuideSectionTypes.JS,
          code: iconTypesSource,
        },
      ],
      props: { OuiIcon },
      demo: <IconTypes />,
    },
  ],
};
