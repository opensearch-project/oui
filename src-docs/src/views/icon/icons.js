/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import {
  OuiFlexGrid,
  OuiFlexItem,
  OuiIcon,
  OuiPanel,
  OuiCopy,
  OuiCodeBlock,
  OuiSpacer,
} from '../../../../src/components';

export const iconTypes = [
  'accessibility',
  'aggregate',
  'alert',
  'analyzeEvent',
  'annotation',
  'anomalyDetection',
  'apmTrace',
  'apps',
  'arrowDown',
  'arrowLeft',
  'arrowRight',
  'arrowUp',
  'asterisk',
  'beaker',
  'bell',
  'bellSlash',
  'bolt',
  'boxesHorizontal',
  'boxesVertical',
  'branch',
  'broom',
  'brush',
  'bug',
  'bullseye',
  'calendar',
  'chatLeft',
  'chatRight',
  'check',
  'checkInCircleEmpty',
  'checkInCircleFilled',
  'cheer',
  'clock',
  'cloudDrizzle',
  'cloudStormy',
  'cloudSunny',
  'color',
  'compute',
  'console',
  'continuityAbove',
  'continuityAboveBelow',
  'continuityBelow',
  'continuityWithin',
  'controlsHorizontal',
  'controlsVertical',
  'copy',
  'copyClipboard',
  'cross',
  'crosshairs',
  'crossInCircleEmpty',
  'crossInCircleFilled',
  'currency',
  'cut',
  'database',
  'document',
  'documentation',
  'documentEdit',
  'documents',
  'dot',
  'download',
  'dockedBottom',
  'dockedDetached',
  'dockedLeft',
  'dockedRight',
  'dockedTakeover',
  'dockedTop',
  'email',
  'empty',
  'eql',
  'eraser',
  'exit',
  'expand',
  'expandMini',
  'exportAction',
  'eye',
  'eyeClosed',
  'faceHappy',
  'faceNeutral',
  'faceSad',
  'filter',
  'flag',
  'fold',
  'folderCheck',
  'folderClosed',
  'folderExclamation',
  'folderOpen',
  'frameNext',
  'framePrevious',
  'fullScreen',
  'fullScreenExit',
  'function',
  'gear',
  'glasses',
  'globe',
  'grab',
  'grabHorizontal',
  'grid',
  'heart',
  'heatmap',
  'help',
  'home',
  'iInCircle',
  'image',
  'importAction',
  'indexClose',
  'indexEdit',
  'indexFlush',
  'indexMapping',
  'indexOpen',
  'indexRuntime',
  'indexSettings',
  'inputOutput',
  'inspect',
  'integrationGeneral',
  'integrationObservability',
  'integrationSearch',
  'integrationSecurity',
  'invert',
  'ip',
  'keyboardShortcut',
  'kqlField',
  'kqlFunction',
  'kqlOperand',
  'kqlSelector',
  'kqlValue',
  'layers',
  'link',
  'list',
  'listAdd',
  'lock',
  'lockOpen',
  'logstashFilter',
  'logstashIf',
  'logstashInput',
  'logstashOutput',
  'logstashQueue',
  'magnifyWithMinus',
  'magnifyWithPlus',
  'magnet',
  'mapMarker',
  'memory',
  'merge',
  'menu',
  'menuDown',
  'menuLeft',
  'menuRight',
  'menuUp',
  'minimize',
  'minus',
  'minusInCircle',
  'minusInCircleFilled',
  'mobile',
  'moon',
  'nested',
  'node',
  'number',
  'offline',
  'online',
  'package',
  'pageSelect',
  'pagesSelect',
  'paperClip',
  'partial',
  'pause',
  'pencil',
  'percent',
  'pin',
  'pinFilled',
  'play',
  'playFilled',
  'plus',
  'plusInCircle',
  'plusInCircleFilled',
  'polygon',
  'popout',
  'power',
  'push',
  'questionInCircle',
  'quote',
  'radius',
  'redeploy',
  'refresh',
  'reporter',
  'returnKey',
  'save',
  'scale',
  'search',
  'securitySignal',
  'securitySignalDetected',
  'securitySignalResolved',
  'shard',
  'share',
  'snowflake',
  'sortable',
  'sortDown',
  'sortLeft',
  'sortRight',
  'sortUp',
  'starEmpty',
  'starEmptySpace',
  'starFilled',
  'starFilledSpace',
  'starMinusEmpty',
  'starMinusFilled',
  'starPlusEmpty',
  'starPlusFilled',
  'stats',
  'stop',
  'stopFilled',
  'stopSlash',
  'storage',
  'string',
  'submodule',
  'symlink',
  'tableOfContents',
  'tableDensityExpanded',
  'tableDensityCompact',
  'tableDensityNormal',
  'tag',
  'tear',
  'temperature',
  'thumbsDown',
  'thumbsUp',
  'timeline',
  'timeslider',
  'training',
  'trash',
  'undeploy',
  'unfold',
  'unlink',
  'user',
  'users',
  'vector',
  'videoPlayer',
  'visArea',
  'visAreaStacked',
  'visBarHorizontal',
  'visBarHorizontalStacked',
  'visBarVertical',
  'visBarVerticalStacked',
  'visBuilder',
  'visBuilderSavedObject',
  'visGauge',
  'visGoal',
  'visLine',
  'visMapCoordinate',
  'visMapRegion',
  'visMetric',
  'visPie',
  'visQueryDQL',
  'visQueryPPL',
  'visQueryPromQL',
  'visQuerySQL',
  'visTable',
  'visTagCloud',
  'visText',
  'visTimelion',
  'visVega',
  'visVisualBuilder',
  'wordWrap',
  'wordWrapDisabled',
  'wrench',
];

export default () => (
  <>
    <OuiCodeBlock language="html" isCopyable paddingSize="m">
      {'<OuiIcon type="alert" />'}
    </OuiCodeBlock>
    <OuiSpacer />
    <OuiFlexGrid direction="column" columns={3}>
      {iconTypes.map((iconType) => (
        <OuiFlexItem key={iconType}>
          <OuiCopy
            display="block"
            textToCopy={iconType}
            afterMessage={`${iconType} copied`}>
            {(copy) => (
              <OuiPanel
                hasShadow={false}
                hasBorder={false}
                onClick={copy}
                paddingSize="s">
                <OuiIcon className="oui-alignMiddle" type={iconType} /> &emsp;{' '}
                <small>{iconType}</small>
              </OuiPanel>
            )}
          </OuiCopy>
        </OuiFlexItem>
      ))}
    </OuiFlexGrid>
  </>
);
