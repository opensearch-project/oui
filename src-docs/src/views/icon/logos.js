/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import {
  OuiFlexGrid,
  OuiFlexItem,
  OuiIcon,
  OuiPanel,
  OuiCodeBlock,
  OuiSpacer,
  OuiCopy,
} from '../../../../src/components';

export const iconTypes = ['logoOpenSearch'].sort();

export default () => (
  <>
    <OuiCodeBlock language="html" isCopyable paddingSize="m">
      {'<OuiIcon type="logoOpenSearch" size="xl" />'}
    </OuiCodeBlock>
    <OuiSpacer />
    <OuiFlexGrid direction="column" columns={3}>
      {iconTypes.map((iconType) => (
        <OuiFlexItem key={iconType}>
          <OuiCopy
            display="block"
            textToCopy={iconType}
            afterMessage={`${iconType} copied`}>
            {(copy) => (
              <OuiPanel
                hasShadow={false}
                hasBorder={false}
                onClick={copy}
                paddingSize="s">
                <OuiIcon
                  className="oui-alignMiddle"
                  type={iconType}
                  size="xl"
                />{' '}
                &emsp; <small>{iconType}</small>
              </OuiPanel>
            )}
          </OuiCopy>
        </OuiFlexItem>
      ))}
    </OuiFlexGrid>
  </>
);
