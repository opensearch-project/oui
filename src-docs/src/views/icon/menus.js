/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import {
  OuiFlexGrid,
  OuiFlexItem,
  OuiIcon,
  OuiPanel,
  OuiCodeBlock,
  OuiCopy,
  OuiSpacer,
} from '../../../../src/components';

const iconTypes = [
  'menusFeatureAdministration',
  'menusFeatureAiFlow',
  'menusFeatureAlerting',
  'menusFeatureAnalyticsWs',
  'menusFeatureAnomalyDetection',
  'menusFeatureDashboards',
  'menusFeatureData',
  'menusFeatureDetectionRules',
  'menusFeatureDevtools',
  'menusFeatureDiscover',
  'menusFeatureEssentialsWs',
  'menusFeatureExperiments',
  'menusFeatureGetStarted',
  'menusFeatureInfo',
  'menusFeatureInfra',
  'menusFeatureIntegrations',
  'menusFeatureJudgements',
  'menusFeatureManage',
  'menusFeatureMaps',
  'menusFeatureModels',
  'menusFeatureNotebooks',
  'menusFeatureNotifications',
  'menusFeatureObservabilityWs',
  'menusFeatureOverview',
  'menusFeatureQuerySets',
  'menusFeatureReports',
  'menusFeatureSearchWs',
  'menusFeatureSecurityCases',
  'menusFeatureSecurityFindings',
  'menusFeatureSecurityWs',
  'menusFeatureServiceMap',
  'menusFeatureServices',
  'menusFeatureSlos',
  'menusFeatureThreatIntel',
  'menusFeatureTicketing',
  'menusFeatureUi',
  'menusSearchConfigurationsln',
];

export default () => (
  <>
    <OuiCodeBlock language="html" isCopyable paddingSize="m">
      {'<OuiIcon type="menusFeatureAdministration" />'}
    </OuiCodeBlock>
    <OuiSpacer />
    <OuiFlexGrid direction="column" columns={3}>
      {iconTypes.map((iconType) => (
        <OuiFlexItem key={iconType}>
          <OuiCopy
            display="block"
            textToCopy={iconType}
            afterMessage={`${iconType} copied`}>
            {(copy) => (
              <OuiPanel
                hasShadow={false}
                hasBorder={false}
                onClick={copy}
                paddingSize="s">
                <OuiIcon className="oui-alignMiddle" type={iconType} /> &emsp;{' '}
                <small>{iconType}</small>
              </OuiPanel>
            )}
          </OuiCopy>
        </OuiFlexItem>
      ))}
    </OuiFlexGrid>
  </>
);
