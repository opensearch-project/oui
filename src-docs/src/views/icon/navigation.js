/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import {
  OuiFlexGrid,
  OuiFlexItem,
  OuiIcon,
  OuiPanel,
  OuiCodeBlock,
  OuiCopy,
  OuiSpacer,
} from '../../../../src/components';

const iconTypes = [
  'featureAdministration',
  'featureAiFlow',
  'featureAlerting',
  'featureAnomalyDetection',
  'featureDashboards',
  'featureData',
  'featureDetectionRules',
  'featureDevtools',
  'featureDiscover',
  'featureExperiments',
  'featureGetStarted',
  'featureInfo',
  'featureInfra',
  'featureIntegrations',
  'featureJudgements',
  'featureManage',
  'featureMaps',
  'featureModels',
  'featureNotebooks',
  'featureNotifications',
  'featureOverview',
  'featureQuerySets',
  'featureReports',
  'featureSecurityCases',
  'featureSecurityFindings',
  'featureServiceMap',
  'featureServices',
  'featureSlos',
  'featureThreatIntel',
  'featureTicketing',
  'featureUi',
  'searchConfigurationsln',
];

export default () => (
  <>
    <OuiCodeBlock language="html" isCopyable paddingSize="m">
      {'<OuiIcon type="featureAdministration" />'}
    </OuiCodeBlock>
    <OuiSpacer />
    <OuiFlexGrid direction="column" columns={3}>
      {iconTypes.map((iconType) => (
        <OuiFlexItem key={iconType}>
          <OuiCopy
            display="block"
            textToCopy={iconType}
            afterMessage={`${iconType} copied`}>
            {(copy) => (
              <OuiPanel
                hasShadow={false}
                hasBorder={false}
                onClick={copy}
                paddingSize="s">
                <OuiIcon className="oui-alignMiddle" type={iconType} /> &emsp;{' '}
                <small>{iconType}</small>
              </OuiPanel>
            )}
          </OuiCopy>
        </OuiFlexItem>
      ))}
    </OuiFlexGrid>
  </>
);
