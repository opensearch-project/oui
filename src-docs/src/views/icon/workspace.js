/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import {
  OuiFlexGrid,
  OuiFlexItem,
  OuiIcon,
  OuiPanel,
  OuiCodeBlock,
  OuiCopy,
  OuiSpacer,
} from '../../../../src/components';

const iconTypes = [
  'wsAnalytics',
  'wsEssentials',
  'wsObservability',
  'wsSearch',
  'wsSecurityAnalytics',
  'wsSelector',
];

export default () => (
  <>
    <OuiCodeBlock language="html" isCopyable paddingSize="m">
      {'<OuiIcon type="wsSelector" />'}
    </OuiCodeBlock>
    <OuiSpacer />
    <OuiFlexGrid direction="column" columns={3}>
      {iconTypes.map((iconType) => (
        <OuiFlexItem key={iconType}>
          <OuiCopy
            display="block"
            textToCopy={iconType}
            afterMessage={`${iconType} copied`}>
            {(copy) => (
              <OuiPanel
                hasShadow={false}
                hasBorder={false}
                onClick={copy}
                paddingSize="s">
                <OuiIcon className="oui-alignMiddle" type={iconType} /> &emsp;{' '}
                <small>{iconType}</small>
              </OuiPanel>
            )}
          </OuiCopy>
        </OuiFlexItem>
      ))}
    </OuiFlexGrid>
  </>
);
