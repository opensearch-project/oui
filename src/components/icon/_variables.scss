/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

$ouiIconLoadingOpacity: .05;

// Modifier naming and colors.
$ouiIconColors: (
  accent: makeGraphicContrastColor($ouiColorAccent),
  danger: makeGraphicContrastColor($ouiColorDanger),
  ghost: $ouiColorGhost,
  primary: makeGraphicContrastColor($ouiColorPrimary),
  secondary: makeGraphicContrastColor($ouiColorSecondary),
  success: makeGraphicContrastColor($ouiColorSuccess),
  subdued: $ouiTextSubduedColor,
  text: $ouiTextColor,
  warning: makeGraphicContrastColor($ouiColorWarning),
  inherit: inherit
);
$ouiIconSizes: (
  small: $ouiSizeM,
  medium: $ouiSize,
  large: $ouiSizeL,
  xLarge: $ouiSizeXL,
  xxLarge: $ouiSizeXXL
);


/* OUI -> EUI Aliases */
$euiIconLoadingOpacity: $ouiIconLoadingOpacity;
$euiIconColors: $ouiIconColors;
$euiIconSizes: $ouiIconSizes;

/* End of Aliases */
