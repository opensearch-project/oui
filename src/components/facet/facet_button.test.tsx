/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render, screen, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { requiredProps } from '../../test/required_props';

import { OuiFacetButton } from './facet_button';
import { OuiIcon } from '../icon';

describe('OuiFacetButton', () => {
  test('is rendered', () => {
    const { container } = render(
      <OuiFacetButton {...requiredProps}>Content</OuiFacetButton>
    );

    expect(container).toMatchSnapshot();
  });

  describe('props', () => {
    describe('isDisabled', () => {
      it('is rendered', () => {
        const { container } = render(
          <OuiFacetButton isDisabled>Content</OuiFacetButton>
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('isLoading', () => {
      it('is rendered', () => {
        const { container } = render(
          <OuiFacetButton isLoading>Content</OuiFacetButton>
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('isSelected', () => {
      it('is rendered', () => {
        const { container } = render(
          <OuiFacetButton isSelected>Content</OuiFacetButton>
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('quantity', () => {
      it('is rendered', () => {
        const { container } = render(
          <OuiFacetButton quantity={60}>Content</OuiFacetButton>
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('icon', () => {
      it('is rendered', () => {
        const { container } = render(
          <OuiFacetButton icon={<OuiIcon type="dot" />}>Content</OuiFacetButton>
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('onClick', () => {
      it('supports onClick', async () => {
        const handler = jest.fn();
        render(<OuiFacetButton onClick={handler}>Content</OuiFacetButton>);

        const user = userEvent.setup();
        await act(async () => {
          await user.click(screen.getByRole('button'));
        });

        expect(handler).toHaveBeenCalledTimes(1);
      });
    });
  });
});
