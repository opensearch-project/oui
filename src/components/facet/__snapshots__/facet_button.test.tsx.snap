// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiFacetButton is rendered 1`] = `
<div>
  <button
    aria-label="aria-label"
    class="ouiFacetButton ouiFacetButton--unSelected testClass1 testClass2"
    data-test-subj="test subject string"
    title="aria-label"
    type="button"
  >
    <span
      class="ouiFacetButton__content"
    >
      <span
        class="ouiFacetButton__text"
        data-text="Content"
      >
        Content
      </span>
    </span>
  </button>
</div>
`;

exports[`OuiFacetButton props icon is rendered 1`] = `
<div>
  <button
    class="ouiFacetButton ouiFacetButton--unSelected"
    title="Content"
    type="button"
  >
    <span
      class="ouiFacetButton__content"
    >
      <span
        class="ouiFacetButton__icon"
        data-ouiicon-type="dot"
      />
      <span
        class="ouiFacetButton__text"
        data-text="Content"
      >
        Content
      </span>
    </span>
  </button>
</div>
`;

exports[`OuiFacetButton props isDisabled is rendered 1`] = `
<div>
  <button
    class="ouiFacetButton ouiFacetButton--unSelected"
    disabled=""
    title="Content"
    type="button"
  >
    <span
      class="ouiFacetButton__content"
    >
      <span
        class="ouiFacetButton__text"
        data-text="Content"
      >
        Content
      </span>
    </span>
  </button>
</div>
`;

exports[`OuiFacetButton props isLoading is rendered 1`] = `
<div>
  <button
    class="ouiFacetButton ouiFacetButton--unSelected"
    disabled=""
    title="Content"
    type="button"
  >
    <span
      class="ouiFacetButton__content"
    >
      <span
        class="ouiFacetButton__text"
        data-text="Content"
      >
        Content
      </span>
      <span
        class="ouiLoadingSpinner ouiLoadingSpinner--medium ouiFacetButton__spinner"
      />
    </span>
  </button>
</div>
`;

exports[`OuiFacetButton props isSelected is rendered 1`] = `
<div>
  <button
    class="ouiFacetButton ouiFacetButton--isSelected"
    title="Content"
    type="button"
  >
    <span
      class="ouiFacetButton__content"
    >
      <span
        class="ouiFacetButton__text"
        data-text="Content"
      >
        Content
      </span>
    </span>
  </button>
</div>
`;

exports[`OuiFacetButton props quantity is rendered 1`] = `
<div>
  <button
    class="ouiFacetButton ouiFacetButton--unSelected"
    title="Content"
    type="button"
  >
    <span
      class="ouiFacetButton__content"
    >
      <span
        class="ouiFacetButton__text"
        data-text="Content"
      >
        Content
      </span>
      <span
        class="ouiNotificationBadge ouiNotificationBadge--medium ouiNotificationBadge--subdued ouiFacetButton__quantity"
      >
        60
      </span>
    </span>
  </button>
</div>
`;
